tensorflow/python/ops/math_grad.py

```diff
@@ -411,6 +417,16 @@ def _SigmoidGrad(op, grad):
     return gen_math_ops._sigmoid_grad(y, grad)
 
 
+@ops.RegisterGradient("SigmoidGrad")
+def _SigmoidGradGrad(op, grad):
+  y = op.outputs[0]
+  with ops.control_dependencies([grad.op]):
+    if y.dtype.is_complex:
+      y = math_ops.conj(y)
+    return (- math.ops.exp(x) * (math.ops.exp(x) - 1) *
+            math_ops.pow(math.ops.exp(x) + 1, -3))
```

Comment:
@girving, one quick question, how do I access tanh_grad and sigmoid_grad in
test?  I'll update the changes of what I had and add the test later.

On Sun, Aug 14, 2016 at 5:28 PM, Geoffrey Irving <notifications@github.com>
wrote:

> In tensorflow/python/ops/math_grad.py
> <https://github.com/tensorflow/tensorflow/pull/3807#discussion_r74711898>:
>
> > @@ -411,6 +417,16 @@ def _SigmoidGrad(op, grad):
> >      return gen_math_ops._sigmoid_grad(y, grad)
> >
> >
> > +@ops.RegisterGradient("SigmoidGrad")
> > +def _SigmoidGradGrad(op, grad):
> > +  y = op.outputs[0]
> > +  with ops.control_dependencies([grad.op]):
> > +    if y.dtype.is_complex:
> > +      y = math_ops.conj(y)
> > +    return (- math.ops.exp(x) * (math.ops.exp(x) - 1) *
> > +            math_ops.pow(math.ops.exp(x) + 1, -3))
>
> It's more parallel with the parentheses.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/3807/files/253c86d7783dca1c844ff77afad5342c40691e66#r74711898>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADzDDHpvD4gwH75k6VBtb7aGUtzmQvJmks5qf7KtgaJpZM4Jj8mC>
> .
>